src/Logger.hpp

```diff
@@ -114,89 +121,88 @@ class Logger {
               "{}Poac {} {}{} ", gray("["), lvlStr, func, gray("]")
           );
         },
-        std::forward<decltype(maybeWriter)>(maybeWriter),
-        std::forward<decltype(msgs)>(msgs)...
+        func, fmt, std::forward<Args>(args)...
     );
   }
 
+  template <typename... Args>
   static void logln(
-      Level level, HeadProcessor auto&& processHead, Writer auto&& writer,
-      Display auto&&... msgs
-  ) noexcept {
-    loglnImpl(
-        std::forward<decltype(writer)>(writer), level,
-        std::forward<decltype(processHead)>(processHead),
-        std::forward<decltype(msgs)>(msgs)...
-    );
-  }
-  static void logln(
-      Level level, HeadProcessor auto&& processHead, Display auto&&... msgs
+      Level level, HeadProcessor auto&& processHead, auto&& head,
+      fmt::format_string<Args...> fmt, Args&&... args
   ) noexcept {
     loglnImpl(
         std::cerr, level, std::forward<decltype(processHead)>(processHead),
-        std::forward<decltype(msgs)>(msgs)...
+        std::forward<decltype(head)>(head), fmt, std::forward<Args>(args)...
     );
   }
 
+  template <typename... Args>
   static void loglnImpl(
       std::ostream& os, Level level, HeadProcessor auto&& processHead,
-      Display auto&&... msgs
+      auto&& head, fmt::format_string<Args...> fmt, Args&&... args
   ) noexcept {
     instance().log(
         os, level, std::forward<decltype(processHead)>(processHead),
-        std::forward<decltype(msgs)>(msgs)..., '\n'
+        std::forward<decltype(head)>(head), fmt, std::forward<Args>(args)...
     );
   }
 
+  template <typename... Args>
   void
   log(std::ostream& os, Level level, HeadProcessor auto&& processHead,
```

Comment:
I like your idea of using `std::endl` here, but now this function name `log` isn't clear as we also have `logln`. Do you have any idea?

Reply:
Both `log` and `logln` function uses header, so there is little reason not to insert newline character.
I think renaming `logln` and `loglnImpl` is sufficient.

Reply:
Let's keep the current code for now. I'll think of this later. Thanks.